lib/time/time.go

```diff
@@ -0,0 +1,454 @@
+}
+
+func location(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.String
+	if err := starlark.UnpackArgs("location", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+	loc, err := gotime.LoadLocation(string(x))
+	if err != nil {
+		return nil, err
+	}
+
+	return starlark.String(loc.String()), nil
+}
+
+func sleep(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.Int
+	if err := starlark.UnpackArgs("sleep", args, kwargs, "x", &x); err != nil {
+		return starlark.None, err
+	}
+	dur, ok := x.Int64()
+	if !ok {
+		return starlark.None, fmt.Errorf("invalid sleep value")
+	}
+
+	gotime.Sleep(gotime.Duration(dur))
+	return starlark.None, nil
+}
+
+func time(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x, location starlark.String
+		format      = starlark.String(gotime.RFC3339)
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location", &location); err != nil {
+		return nil, err
+	}
+
+	if location == "" {
+		t, err := gotime.Parse(format.String(), x.String())
+		if err != nil {
+			return nil, err
+		}
+		return Time(t), nil
+	}
+
+	loc, err := gotime.LoadLocation(location.String())
+	if err != nil {
+		return nil, err
+	}
+	t, err := gotime.ParseInLocation(format.String(), x.String(), loc)
+	if err != nil {
+		return nil, err
+	}
+	return Time(t), nil
+}
+
+func fromtimestamp(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x starlark.Int
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+
+	i, err := strconv.ParseInt(x.String(), 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	t := gotime.Unix(i, 0)
+
+	return Time(t), nil
+}
+
+func now(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	return Time(NowFunc()), nil
+}
+
+// Duration is a starlark representation of a duration
+type Duration gotime.Duration
+
+// String implements the Stringer interface
+func (d Duration) String() string { return gotime.Duration(d).String() }
+
+// Type returns a short string describing the value's type.
+func (d Duration) Type() string { return "duration" }
+
+// Freeze renders Duration immutable. required by starlark.Value interface
+// because duration is already immutable this is a no-op
+func (d Duration) Freeze() {}
+
+// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y)
+// required by starlark.Value interface
+func (d Duration) Hash() (uint32, error) { return hashString(d.String()), nil }
+
+// Truth returns the truth value of an object required by starlark.Value interface
+func (d Duration) Truth() starlark.Bool { return d > 0 }
```

Comment:
d != 0.  It's a signed quantity (though I'm not sure exactly what a negative duration means).